### PR TITLE
test cases complete

### PR DESCRIPTION
**Pull request by BernardHan**
#914

### CODE REVIEW

File: src/network/graphql/__mocks__/queries.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/** This is a mock module of ../queries.js,
```
Comment (dwalintukan) [inline src/network/graphql/__mocks__/queries.js]: can we rename the folder for `__mocks__` to just `mocks`? or is it needed to have the underscores?

File: src/stores/AllEventsStore.test.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import { Routes } from 'constants';
+jest.mock('../network/graphql/queries'); // block and manually mock our backend
```
Comment (dwalintukan) [inline src/stores/AllEventsStore.test.js]: move under imports?

File: src/stores/AllEventsStore.test.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+const localStorageMock = {
+  getItem: jest.fn(),
+  setItem: jest.fn(),
+  clear: jest.fn(),
+};
+const navigatorMock = {};
+global.navigator = navigatorMock;
+global.localStorage = localStorageMock;
+
+/** test */
```
Comment (dwalintukan) [inline src/stores/AllEventsStore.test.js]: remove unnecessary comments

File: src/stores/AllEventsStore.test.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  }; // mock the appstore
+  beforeEach(() => {
+    store = new AllEventsStore(app); // create a new instance before each test case
+  });
+
+  /** all following test cases target specifically to the mock backend, eg. network/graphql/__mocks__/queries.js */
+  it('Init', async () => {
+    const curListNum = store.list.length;
+    expect(curListNum).toBe(0);
+    expect.assertions(4);
```
Comment (dwalintukan) [inline src/stores/AllEventsStore.test.js]: curious, what does this do?

File: src/stores/AllEventsStore.test.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    await store.init();
+    expect(app.ui.location).toBe(Routes.ALL_EVENTS); // test webpage location
+    expect(store.list.length).toBe(curListNum + store.limit); // test if list got new cards
+    expect(store.loading).toBe(false); // test runInAction changes
```
Comment (dwalintukan) [inline src/stores/AllEventsStore.test.js]: can remove unnecessary comments. these are self-explanatory by the `expect.toBe` as they read like sentences

File: src/stores/AllEventsStore.test.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    expect(store.list.length).toBe(store.limit);
+    expect(store.loadingMore).toBe(false);
+    await store.loadMoreEvents();
+    expect(store.list.length).toBe((curListNum + store.limit) - 2); // mock backend only has 46 events in total
+    expect(store.loadingMore).toBe(false);
+    expect(store.hasMore).toBe(false); // no events left in backend
+    expect(store.skip).toBe(curSkip + store.limit);
+  });
+
+  /** following test cases test the reaction function */
```
Comment (dwalintukan) [inline src/stores/AllEventsStore.test.js]: again, unnecessary since we can tell by the test name

File: src/stores/AllEventsStore.test.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+  it('Reaction SortBy', () => {
+    app.ui.location = Routes.ALL_EVENTS;
```
Comment (dwalintukan) [inline src/stores/AllEventsStore.test.js]: would be good to check the starting state of the list length before the reaction

File: src/stores/AllEventsStore.test.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    setTimeout(() => {
+      expect.assertions(2);
+      expect(store.app.sortBy).toBe('Yo');
+      expect(store.list.length).toBe(store.limit);
+    }, 500);
+  });
+
+  it('Reaction Wallet Addr', () => {
+    app.ui.location = Routes.ALL_EVENTS;
+    app.wallet.addresses = 1;
```
Comment (dwalintukan) [inline src/stores/AllEventsStore.test.js]: when testing the reaction, we should test with how a real change should be. this should be an array of objects.

File: src/stores/AllEventsStore.test.js (hunk 2 above)
Comment (BernardHan) [reply src/stores/AllEventsStore.test.js]: Initially I thought this has to be before the import module that uses backend, but now it seems unnecessary. Moved.

File: src/network/graphql/__mocks__/queries.js (hunk 1 above)
Comment (BernardHan) [reply src/network/graphql/__mocks__/queries.js]: It seems to be a must. Google & Jest official tells me nothing about changing the folder name.

File: src/stores/AllEventsStore.test.js (hunk 4 above)
Comment (BernardHan) [reply src/stores/AllEventsStore.test.js]: Line 31 & 32 is trying to make sure the list is empty before init.
Line 33 says this test case should pass 4 assertions. This line is unnecessary for this test case but good to have. But in general, it's useful in asynchronous tests.
For example,
`it('calls the API and throws an error', async () => {
  expect.assertions(2);
  try {
    await login('email', 'password');
  } catch (error) {
    expect(error.name).toEqual('Unauthorized');
    expect(error.status).toEqual(401);
  }
});`
Without it, this test case will pass even when login successful.

Or 
`test('doAsync calls both callbacks', () => {
  expect.assertions(2);
  function callback1(data) {
    expect(data).toBeTruthy();
  }
  function callback2(data) {
    expect(data).toBeTruthy();
  }

  doAsync(callback1, callback2);
});`
Here it ensures that both callbacks are actually called.

File: src/stores/AllEventsStore.test.js (hunk 5 above)
Comment (BernardHan) [reply src/stores/AllEventsStore.test.js]: removed

File: src/stores/AllEventsStore.test.js (hunk 8 above)
Comment (BernardHan) [reply src/stores/AllEventsStore.test.js]: Are you saying I need to check the list retrieved from the mocked backend should be the same as what I stuff in the mocked backend?

File: src/network/graphql/__mocks__/queries.js (hunk 1 above)
Comment (dwalintukan) [reply src/network/graphql/__mocks__/queries.js]: does the tests work if you do rename it?

File: src/stores/AllEventsStore.test.js (hunk 4 above)
Comment (dwalintukan) [reply src/stores/AllEventsStore.test.js]: gotcha, its just saying how many asserts are required.

File: src/network/graphql/__mocks__/queries.js (hunk 1 above)
Comment (BernardHan) [reply src/network/graphql/__mocks__/queries.js]: no it does not

**Pull request merged**